### PR TITLE
USWDS-Site: Add tooltip accessibility tests page

### LINKED ISSUE #2525: CC a11y: publish Tooltip checklist - create PR
## Summary
Publish the tooltip accessibility checklist items created in issue #2524.

## Related files
- [Critical Checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?usp=sharing) (Google Doc :lock:)
- [Creating accessibility checklists - process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit#heading=h.fa3sv1ray42g) (Google Doc :lock:)

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary

Added accessibility tests page for the tooltip component.

> [!Important]
> We should update changelog dates before merge.

## Related issue

Closes #2525

## Preview link

- [Tooltip accessibility tests page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-a11y-tests-tooltip/components/tooltip/accessibility-tests/)
- [Tooltip main component page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-a11y-tests-tooltip/components/tooltip/)

## Resources
- [Creating accessibility checklists process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit) (Google docs 🔒)
- [Accessibility checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit#gid=97481116) (Google sheets 🔒)

## Testing and review
1. Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
1. Confirm that the main component page links to the accessibility tests page in the side navigation.
1. Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
1. Confirm that the main component page shows the accessibility tests summary.
1. Check that accessibility tests page provides the correct counts for each test status type.
1. Confirm the test checklist summaries and data are accurate.  
1. Confirm no visual issues.
1. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

### CODE REVIEW

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [inline _data/accessibility-tests/tooltip.yml:43]: @amycole501 @alex-hull 
I dug into this [criteria](https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) a bit more and it looks like there are three criteria (shown below). It looks like tooltip passes "Persistent", but fails "Dismissable" and "Hoverable". Right now, the only test we have related to 1.4.13 is for "Dismissable". Is it reasonable to re-write this checklist item to include "Hoverable" and "Persistent"? If so, could we mark it as "passed with exceptions"?

> Dismissible
> A [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://www.w3.org/TR/WCAG21/#dfn-input-error) or does not obscure or replace other content;
>
> Hoverable
> If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
>
> Persistent
> The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [inline _data/accessibility-tests/tooltip.yml:5]: @sarah-sch @finekatie 
Sometimes we call this "tooltip" and other times "tool tip". Is it ok to use both?
```suggestion
  - summary: Color is not the only method used to communicate the purpose of the tooltip.
```

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [inline _data/accessibility-tests/tooltip.yml:78]: @amycole501 @alex-hull @sarah-sch @finekatie 
I updated the summary and test status for 1.3.1 according to [this conversation](https://github.com/uswds/uswds/issues/5891#issuecomment-2088981813). I also removed the test for 4.1.2. Please let me know if you want any changes here!

**Review by amyleadem — COMMENTED** [review]
This is ready for review. I added comments with some open questions below. Please let me know if you need any changes.

File: _data/changelogs/component-tooltip-accessibility.yml
Comment (amyleadem) [inline _data/changelogs/component-tooltip-accessibility.yml:5]: > [!Important]
> We should update changelog dates before merge.

**Review by amycole501 — APPROVED** [review]
The page reads well and covers the success criteria we tested, all good.

File: _data/accessibility-tests/tooltip.yml
Comment (finekatie) [reply _data/accessibility-tests/tooltip.yml:5]: I'd suggest using the same term throughout for consistency. The only time I'd change it is when there's a concern about meaning confusion.

**Review by finekatie — APPROVED** [review]
- [x] Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
- [x] Confirm that the main component page links to the accessibility tests page in the side navigation.
- [x] Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
- [x] Confirm that the main component page shows the accessibility tests summary.
- [x] Check that accessibility tests page provides the correct counts for each test status type.
- [x] Confirm no visual issues.
- [x] Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

-  Confirm the test checklist summaries and data are accurate. - relying on Amy C and Alex H expertise on this one.

otherwise LGM!!

File: _data/accessibility-tests/tooltip.yml
Comment (alex-hull) [reply _data/accessibility-tests/tooltip.yml:43]: I think that is reasonable. Would it make sense to split these into three different prompts, but still have them related to the one criteria? I think that would just help it look cleaner. I also cannot remember how we have done ones with multiple considerations in the past.

**Review by amycole501 — APPROVED** [review]
The three new checks added (two failing) read well to me. The page overall reads well. I couldn't identify any changes needed.

**Review by alex-hull — APPROVED** [review]
I think the fails look good (until we mark them as pass)!

File: _data/accessibility-tests/tooltip.yml
Comment (sarah-sch) [inline _data/accessibility-tests/tooltip.yml:37]: Recommending a slight revision: Active tooltips close only when you move focus away from the tooltip trigger or press the `escape` key.

File: _data/accessibility-tests/tooltip.yml
Comment (sarah-sch) [inline _data/accessibility-tests/tooltip.yml:57]: Format `escape` with backticks instead of quotes.

**Review by sarah-sch — CHANGES_REQUESTED** [review]
@amyleadem - I'm recommending a couple of tweaks. Thanks!

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: Thanks! Updated in [0c9c00e](https://github.com/uswds/uswds-site/pull/2646/commits/0c9c00e2f25f7fd9b408d8807ff55992aa5cbd07)

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: Updated in [0c9c00e](https://github.com/uswds/uswds-site/pull/2646/commits/0c9c00e2f25f7fd9b408d8807ff55992aa5cbd07)

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [inline _data/accessibility-tests/tooltip.yml:61]: @amycole501 @alex-hull @sarah-sch 

I think this should be re-written a bit since it doesn't seem to be about closing it with a mouse. I borrowed some language from [1.4.13](https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) for this suggestion. I also wonder if this should be moved to "general". Let me know what you think. 

```suggestion
  - summary: Active tooltips can be dismissed without moving the mouse or keyboard focus. 
    summary_additional: When you open a tooltip with a mouse, you can use the `escape` key to close it.
    test_status: fail
    github_issue_number: 5909
    github_issue_repo: uswds
    test_type: general
```

Comment (sarah-sch) [reply _data/accessibility-tests/tooltip.yml:61]: @amyleadem - the language looks good to me. I will leave it to @amycole501 and @alex-hull to confirm that the information is correct.

Comment (amycole501) [reply _data/accessibility-tests/tooltip.yml:61]: The new wording does make for a simpler-to-follow test. My only slight concern is that the person knows that even if they hover over the element to activate the tooltip, the tooltip should be persistent even if you move the mouse. BUT, I think that your wording change still incorporates that into the test so we are probably good.

Also, good catch on moving this to general since both a keyboard and a mouse are involved.

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: I split these out into three separate prompts in [252d4b6](https://github.com/uswds/uswds-site/pull/2646/commits/252d4b6afebb71649c5604109ea0cdd6fd6e90df). They are currently marked as "fail" but fixes are in the works and we are hoping to get those out ASAP for 3.8.1.

File: _components/tooltip/accessibility-tests.md
Comment (mahoneycm) [inline _components/tooltip/accessibility-tests.md:8]: Adding component here for consistency 
```suggestion
lead: Any USWDS tooltip component should pass these manual accessibility tests.
```

File: _data/accessibility-tests/tooltip.yml
Comment (mahoneycm) [inline _data/accessibility-tests/tooltip.yml:2]: Our tests currently show two `AA` failures. Is there a reason we aren't marking this as a failure for now?

**Review by mahoneycm — CHANGES_REQUESTED** [review]
Looking good! One update and one question for ya

**Review by sarah-sch — APPROVED** [review]
Approving Amy L.'s tweaks to language for 1.4.13 tests added during review process.

**Comment by amyleadem** [comment]
Putting this effort on hold until the fixes uswds/uswds#5919 and uswds/uswds#5909 are merged.

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml:2]: I left this as "passing" because the fixes are in the works and should be published in 3.8.1. For consistency, I just updated the test status for these failing items to `pass` since it looks like the fixes are in a good state to be merged in 3.8.1. I also added a note at the top of the PR description saying that this PR should not be merged until these test items are indeed passing.

File: _components/tooltip/accessibility-tests.md
Comment (amyleadem) [reply _components/tooltip/accessibility-tests.md]: Good catch. Added in [1ced497](https://github.com/uswds/uswds-site/pull/2646/commits/1ced49732cf1854a229de37d3eb993fa95c8c726).

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: I think this is a good note. I tried adding a bit of clarity to the summary in [bdb277a](https://github.com/uswds/uswds-site/pull/2646/commits/bdb277aeb9f784c271fb821574458f0538571e48).

"Active tooltips can be dismissed without moving the mouse or keyboard focus." --> 
"Active tooltips can be dismissed without moving the mouse or keyboard focus _away from the tooltip trigger_."

Let me know what you think!

**Comment by amyleadem** [comment]
Everyone - I have moved this out of hold now that the tooltip fixes in 3.8.1 have been published. Please let me know if you need any changes. If not, I'll move it to fed final.

**Review by amycole501 — APPROVED** [review]
Looks good!

File: _data/accessibility-tests/tooltip.yml
Comment (mahoneycm) [inline _data/accessibility-tests/tooltip.yml:32]: Hover functionality was added in `3.8.1`, we should update here

```suggestion
    version_tested: 3.8.1
```

File: _data/accessibility-tests/tooltip.yml
Comment (mahoneycm) [inline _data/accessibility-tests/tooltip.yml:46]: Escape functionality was added in `3.8.1` so we should update here
```suggestion
    version_tested: 3.8.1
```

File: _data/accessibility-tests/tooltip.yml
Comment (mahoneycm) [inline _data/accessibility-tests/tooltip.yml:85]: This is currently marked as a failure on the checklist. Do we want to update this to reflect that?

```suggestion
    test_status: fail
```

**Review by mahoneycm — CHANGES_REQUESTED** [review]
Requested possible changes related to test failures and behaviors introduced in `3.8.1`

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml:85]: Good note. We decided in [this conversation](https://github.com/uswds/uswds-site/pull/2646#discussion_r1587813645) and [this conversation](https://github.com/uswds/uswds/issues/5891#issuecomment-2088981813) to change up the test status and summary. I added a comment to this summary in the spreadsheet top request it be updated there as well. @amycole501 and @alex-hull, can you confirm this change and update the spreadsheet to match?

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: Goooood catch. Updated in [1dd578e](https://github.com/uswds/uswds-site/pull/2646/commits/1dd578eb4c656dc20e8390134c91abaf513011b2)

File: _data/accessibility-tests/tooltip.yml
Comment (amyleadem) [reply _data/accessibility-tests/tooltip.yml]: Goooood catch. Updated in [1dd578e](https://github.com/uswds/uswds-site/pull/2646/commits/1dd578eb4c656dc20e8390134c91abaf513011b2)

File: _data/accessibility-tests/tooltip.yml
Comment (alex-hull) [reply _data/accessibility-tests/tooltip.yml:85]: I updated the statuses, and like the simplified status for 1.3.1 and 4.1.2

**Review by mahoneycm — APPROVED** [review]
LGTM!

**Review by thisisdano — APPROVED** [review]

**Pull request merged**
